From 6422e08df16f59380449264127e362e185f29ff0 Mon Sep 17 00:00:00 2001
From: Jeffy Chen <jeffy.chen@rock-chips.com>
Date: Mon, 3 Oct 2022 21:03:04 +0800
Subject: [PATCH 83/98] desktop-shell: Fix crash when hotplugging screens

1/ Force allowing respawn.
2/ Panel's output might not be available when committing.

Signed-off-by: Jeffy Chen <jeffy.chen@rock-chips.com>
---
 desktop-shell/shell.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/desktop-shell/shell.c b/desktop-shell/shell.c
index dc0aeb8..f3481f2 100644
--- a/desktop-shell/shell.c
+++ b/desktop-shell/shell.c
@@ -3006,6 +3006,10 @@ panel_committed(struct weston_surface *es,
 	 * transform is updated here. */
 	weston_view_update_transform(view);
 
+	/* The output might be unavaiable */
+	if (!view->output)
+		return;
+
 	get_panel_size(shell, view, &width, &height);
 	switch (shell->panel_position) {
 	case WESTON_DESKTOP_SHELL_PANEL_POSITION_TOP:
@@ -4343,6 +4347,8 @@ check_desktop_shell_crash_too_early(struct desktop_shell *shell)
 	if (clock_gettime(CLOCK_MONOTONIC, &now) < 0)
 		return false;
 
+	/* HACK: The shell might be crashed too early when hotplugging */
+#if 0
 	/*
 	 * If the shell helper client dies before the session has been
 	 * up for roughly 30 seconds, better just make Weston shut down,
@@ -4358,6 +4364,7 @@ check_desktop_shell_crash_too_early(struct desktop_shell *shell)
 
 		return true;
 	}
+#endif
 
 	return false;
 }
-- 
2.20.1

